src/RequestHandler.cpp

```diff
@@ -1,29 +1,58 @@
 #include "RequestHandler.h"
 #include "ConnectKafka.h"
 #include <thread>
-int RequestHandler::init() {
-  std::string Topic = "MULTIPART_events";
 
-  std::cout << "_________" << std::endl
-            << KafkaConnection->getAllTopics() << std::endl
-            << "___________" << std::endl;
-
-  std::vector<std::string> ToSubscribe;
-  if (KafkaConnection->checkIfTopicExists(Topic)) {
-    ToSubscribe.push_back(Topic);
-    for (auto &SingleStruct : KafkaConnection->getHighAndLowOffsets(Topic)) {
-      std::cout << SingleStruct.PartitionId << " " << SingleStruct.LowOffset
-                << " " << SingleStruct.HighOffset << std::endl;
-    }
-    subscribeConsumeAtOffset(Topic, 6000);
-    std::cout << "##################\n##################\n##################\n#"
-                 "#################\n";
-    subscribeConsumeNLastMessages(Topic, 3);
-  } else
-    std::cout << "No such topic" << std::endl;
+int RequestHandler::init(UserArgumentStruct UserArguments) {
+  checkAndRun(UserArguments);
   return 0;
 }
 
+// check whether arguments passed match any methods
+void RequestHandler::checkAndRun(UserArgumentStruct UserArguments) {
+  try {
+    // check input if ConsumerMode chosen
+    if (UserArguments.ConsumerMode && !UserArguments.MetadataMode)
+      checkConsumerModeArguments(UserArguments);
+
+    // check input if MetadataMode chosen
+    else if (!UserArguments.ConsumerMode && UserArguments.MetadataMode)
+      checkMetadataModeArguments(UserArguments);
+    // no MetadataMode or ConsumerMode chosen
+    else
+      throw ArgumentsException(
+          "Program can run in one and only one mode: --consumer or --metadata");
+  } catch (ArgumentsException E) {
+    E.printException();
+  } catch (std::exception E) {
+    std::cout << E.what() << std::endl;
+  }
+}
+
+void RequestHandler::checkConsumerModeArguments(
+    UserArgumentStruct UserArguments) {
+  if ((UserArguments.GoBack > -2 && UserArguments.OffsetToStart > -2) ||
+      (UserArguments.GoBack == -2 && UserArguments.OffsetToStart == -2))
+    throw ArgumentsException("Program must take one and only one of the "
+                             "arguments: \"--go\",\"--Offset\"");
+  else {
+    UserArguments.OffsetToStart > -2
+        ? subscribeConsumeAtOffset(UserArguments.Name,
+                                   UserArguments.OffsetToStart)
+        : subscribeConsumeNLastMessages(UserArguments.Name,
+                                        UserArguments.GoBack);
+  }
+}
+
+void RequestHandler::checkMetadataModeArguments(
+    UserArgumentStruct UserArguments) {
+  if (!UserArguments.ShowAllTopics && !UserArguments.ShowPartitionsOffsets)
+    throw ArgumentsException("No action specified for \"--list\" mode");
+  else if (UserArguments.ShowAllTopics)
+    std::cout << KafkaConnection->getAllTopics() << std::endl;
```

Comment:
`std::endl` prints a new line and immediately flushes the stream. In this case, and in most cases, you don't really need to flush immediately so it is better to be in the habit of using "\n" and avoiding the overhead of additional flushes.